examples/specs_integration/src/sync/entity_create.rs

```diff
@@ -0,0 +1,122 @@
+use godot::*;
+use specs::prelude::*;
+use specs::world::Index;
+use std::collections::HashMap;
+
+use crate::helper::instance_scene;
+use crate::sync::{GdSceneLoc, GdSpatial, Template3dEntity, Template3dName};
+
+pub struct GdEntityCreation<'a> {
+    pub grandparent: Spatial,
+    pub templates: &'a HashMap<Template3dName, Template3dEntity>,
+    pub spatials: &'a mut HashMap<Index, Spatial>,
+
+    pub default_parent: Spatial,
+}
+
+unsafe impl<'a> Sync for GdEntityCreation<'a> {}
+unsafe impl<'a> Send for GdEntityCreation<'a> {}
+
+impl<'a> System<'a> for GdEntityCreation<'a> {
+    type SystemData = (
+        Entities<'a>,
+        WriteStorage<'a, GdSpatial>,
+        ReadStorage<'a, GdSceneLoc>,
+    );
+
+    fn run(&mut self, (entities, mut gd_spatials, gd_scenelocs): Self::SystemData) {
+        for (entity, gd_sceneloc, gd_spatial_entry) in
+            (&entities, &gd_scenelocs, gd_spatials.entries()).join()
+        {
+            let GdEntityCreation {
+                grandparent,
+                templates,
+                spatials,
+                default_parent: _,
+            } = self;
+
+            unsafe {
+                let parent = if let Some(parent) = grandparent.find_node(
+                    GodotString::from_str(gd_sceneloc.parent_name),
+                    false,
+                    false,
+                ) {
+                    parent
+                } else {
+                    // TODO: Create parent here.
+
+                    let mut new_parent = Spatial::new()
+                        .cast()
+                        .expect("Spatial is a hardcoded subclass of Node! WTF!?");

```

Comment:
Looks like a parent node is already created here. Is this still TODO? If it is, what is to be done exactly?